static/profile/js/profile.js

```diff
@@ -0,0 +1,515 @@
+      if (i>0 && toMinutesFromMidnight(c_profile[array][i-1].time) >= time*30) { return false; } 
+      return true;
+    }
+ 
+    function addSingleLine(e,i) {
+      var tr = $('<tr>');
+      var select = $('<select>').attr('class','pe_selectabletime').attr('id',e.prefix+'_from_'+i);
+      var lowesttime=-1;
+      var selectedValue;
+      for (var t=0;t<48;t++) {
+        if (shouldAddTime(i,t,e.array)) { 
+          if (lowesttime === -1) { lowesttime = t*30; }
+          var label = serverSettings.settings.timeFormat==='24' ? mmoltime[t] : mgtime[t];
+          var value = toTimeString(t*30);
+          if (toMinutesFromMidnight(c_profile[e.array][i].time) === t*30) {
+            selectedValue = value;
+          }
+          select.append('<option value="' + value + '">' + label + '</option>');
+        }
+      }
+      select.val(selectedValue);
+
+      tr.append($('<td>').append('From: ').append(select));
+      tr.append($('<td>').append(e.label).append($('<input type="text">').attr('id',e.prefix+'_val_'+i).attr('value',c_profile[e.array][i].value)));
+      var icons_td = $('<td>').append($('<img>').attr('class','addsingle').attr('style','cursor:pointer').attr('title','Add new interval before').attr('src',icon_add).attr('array',e.array).attr('pos',i));
+      if (c_profile[e.array].length>1) {
+        icons_td.append($('<img>').attr('class','delsingle').attr('style','cursor:pointer').attr('title','Delete interval').attr('src',icon_remove).attr('array',e.array).attr('pos',i));
+      }
+      tr.append(icons_td);
+        
+      if (lowesttime>toMinutesFromMidnight(c_profile[e.array][i].time)) { 
+        c_profile[e.array][i].time = toTimeString(lowesttime);
+      }
+      return tr[0].outerHTML;
+    }
+    
+    // Fill dropdown boxes
+    [{prefix:'pe_basal', array:'basal', label:'Rate: '},
+     {prefix:'pe_ic', array:'carbratio', label:'IC: '},
+     {prefix:'pe_isf', array:'sens', label:'ISF: '}
+    ].forEach(function (e) {
+      var html = '<table>';
+      for (var i=0; i<c_profile[e.array].length; i++) {
+        html += addSingleLine(e,i);
+      }
+      html += '<tr><td></td><td></td><td><img class="addsingle" style="cursor:pointer" title="Add new interval before" src="'+icon_add+'" array="'+e.array+'" pos="'+i+'" href="#"></td></tr>';
+      html += '</table>';
+      $('#'+e.prefix+'_placeholder').html(html);
+    });
+    
+    $('.addsingle').click(function addsingle_click() {
+      var array = $(this).attr('array');
+      var pos = $(this).attr('pos');
+      GUIToObject(); 
+      c_profile[array].splice(pos,0,{time:'00:00',value:0});
+      return fillTimeRanges();
+    });
+    
+    $('.delsingle').click(function delsingle_click() {
+      var array = $(this).attr('array');
+      var pos = $(this).attr('pos');
+      GUIToObject(); 
+      c_profile[array].splice(pos,1);
+      c_profile[array][0].time = '00:00';
+      return fillTimeRanges();
+    });
+
+    function addBGLine(i) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/859032400/issues/source?bid=2452379&fileBranchId=2815005#l291)